File: Leetcode第1047题.py
class Solution(object):
    def removeDuplicates(self, s):
        """
        :type s: str
        :rtype: str
        """
        temp=[]
        for i in s:
            if len(temp)!=0 and i==temp[-1]:
                temp.pop()
            else:
                temp.append(i)
        return ''.join(temp)

File: Leetcode第1047题.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        stack<char> stk;
        for(int i=0;i<s.size();++i){
            if(!stk.empty()&&s[i]==stk.top()){
                stk.pop();
                s.erase(i-1,2);
                i-=2;
            }
            else{stk.push(s[i]);}
        }
        return s;
    }
};
